docs/standard/microservices-architecture/architect-microservice-container-applications/direct-client-to-microservice-communication-versus-the-API-Gateway-pattern.md

```diff
@@ -0,0 +1,121 @@
+
+A direct client-to-microservice communication architecture could be good enough for a small microservice-based application, especially if the client app is a server-side web application like an ASP.NET MVC app. However, when you build large and complex microservice-based applications (for example, when handling dozens of microservice types), and especially when the client apps are remote mobile apps or SPA web applications, that approach faces a few issues.
+
+Consider the following questions when developing a large application based on microservices:
+
+-   *How can client apps minimize the number of requests to the backend and reduce chatty communication to multiple microservices?*
+
+Interacting with multiple microservices to build a single UI screen increases the number of roundtrips across the Internet. This increases latency and complexity on the UI side. Ideally, responses should be efficiently aggregated in the server side—this reduces latency, since multiple pieces of data come back in parallel and some UI can show data as soon as it is ready.
+
+-   *How can you handle cross-cutting concerns such as authorization, data transformations, and dynamic request dispatching?*
+
+Implementing security and cross-cutting concerns like security and authorization on every microservice can require significant development effort. A possible approach is to have those services within the Docker host or internal cluster, in order to restrict direct access to them from the outside, and to implement those cross-cutting concerns in a centralized place, like an API Gateway.
+
+-   *How can client apps communicate with services that use non-Internet-friendly protocols?*
+
+Protocols used on the server side (like AMQP or binary protocols) are usually not supported in client apps. Therefore, requests must be performed through protocols like HTTP/HTTPS and translated to the other protocols afterwards. A *man-in-the-middle* approach can help in this situation.
+
+-   *How can you shape a façade especially made for mobile apps? *
+
+The API of multiple microservices might not be well designed for the needs of different client applications. For instance, the needs of a mobile app might be different than the needs of a web app. For mobile apps, you might need to optimize even further so that data responses can be more efficient. You might do this by aggregating data from multiple microservices and returning a single set of data, and sometimes eliminating any data in the response that is not needed by the mobile app. And, of course, you might compress that data. Again, a façade or API in between the mobile app and the microservices can be convenient for this scenario.
```

Comment:
all this content is styled as italic. not sure if this is just a glitch on GH rendering

Reply:
I don't see that in Italic when editing in VS Code. It must be a glitch on GitHub

Reply:
I haven't reviewed all the other changes yet, but I think this is because of the space between the ? and the * on line 46. I caught a lot of formatting errors like that on the original translation from doc to MD.

Reply:
👍 That's right. That space shouldn't be in between the ? and the *. I just removed the " " and pushed the minor change into the PR.